src/apps/integrations/service/integrations.py

```diff
@@ -45,30 +46,65 @@ async def disable_integration(self, query: QueryParams):
             workspace.integrations = None
         await UserWorkspaceCRUD(self.session).save(workspace)
 
-    async def create_integration(
-        self,
-        type: str,
-        params: IntegrationsCreate,
-    ) -> LorisProjects:
-        match type:
+    async def create_integration(self, newIntegration: Integration) -> Integration:
+        match newIntegration.integration_type:
             case AvailableIntegrations.LORIS:
-                return await self._create_integration(type, params)
```

Comment:
I did not want to use cast, I had to do it because mypy was not happy.
Not sure how to solve the issue I get using Union and mypy without a cast.

Reply:
I like the Factory class idea, I will read about it.

Reply:
any updates here @david-montano-metalab 

I believe mypy is already validating params prior to `cast` and is somewhat safe to use in this scenario

Reply:
After some discussion and attempts I feel it will be better to create a ticket to address the Factory class solution in another ticket.
I agree `mypy` helps validating params in this case.
The Factory class attempt I was writing ended up depending on a user and a session which does not seem reasonable for me so maybe I am tackling it the wrong way.

I will create a JIRA ticket to go back to this implementation afterwards: https://mindlogger.atlassian.net/browse/M2-7909?atlOrigin=eyJpIjoiNzA2Y2QyZWRhZTBjNDRmNDk0MDg1YWQ2YTdjNmQ2YzciLCJwIjoiaiJ9